File: src/test/programmers/level1/GreatestCommonDivisorAndLeastCommonMultiple.java
package test.programmers.level1;

import java.util.Arrays;

public class GreatestCommonDivisorAndLeastCommonMultiple {
    static int[] solution(int n, int m) {
        int[] answer = {};
        int min = n;
        int max = m;

        if(min > max) {
            int temp = min;
            min = max;
            max = temp;
        }

        int gcd = calcGCD(min, max);
        int lcm = (min * max) / gcd;

        answer = new int[]{gcd, lcm};

        return answer;
    }

    static int calcGCD(int min, int max) {
        //유클리드 호제법
        while(true) {
            if(max % min == 0) {
                return min;
            } else {
                int temp = max % min;
                max = min;
                min = temp;
            }
        }
    }
}

File: src/test/programmers/python/level1/GreatestCommonDivisorAndLeastCommonMultiple.py
def solution(n, m):
    answer = []
    a, b = min(n, m), max(n, m)
    answer.append(gcd(a, b))
    answer.append(int(a * b / gcd(a, b)))
    return answer

def gcd(n, m):
    while True:
        if m % n == 0:
            return n
            break
        else:
            temp = m % n
            m = n
            n = temp
